javalib/src/main/scala/java/time/YearMonth.scala

```diff
@@ -0,0 +1,246 @@
+    case DECADES   => plusYears(MathJDK8Bridge.multiplyExact(amount, 10))
+    case CENTURIES => plusYears(MathJDK8Bridge.multiplyExact(amount, 100))
+    case MILLENNIA => plusYears(MathJDK8Bridge.multiplyExact(amount, 1000))
+
+    case ERAS      =>
+      val era = getLong(ERA)
+      `with`(ERA, MathJDK8Bridge.addExact(era, amount))
+
+    case _: ChronoUnit =>
+      throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit)
+
+    case _ => unit.addTo(this, amount)
+  }
+
+  def plusYears(years: Long): YearMonth = {
+    val newYear = YEAR.checkValidIntValue(year + years)
+    if (newYear == year) this
+    else new YearMonth(newYear, month)
+  }
+
+  def plusMonths(months: Long): YearMonth = {
+    if (months == 0) this
+    else {
+      val newProlepticMonth = prolepticMonth + months
+      val newYear = MathJDK8Bridge.floorDiv(newProlepticMonth, 12)
+      val newMonth = MathJDK8Bridge.floorMod(newProlepticMonth, 12) + 1
+      new YearMonth(
+        YEAR.checkValidIntValue(newYear),
+        MONTH_OF_YEAR.checkValidIntValue(newMonth)
+      )
+    }
+  }
+
+  override def minus(amount: Long, unit: TemporalUnit): YearMonth =
+    if (amount != Long.MinValue) plus(-amount, unit)
+    else plus(Long.MaxValue, unit).plus(1, unit)
+
+  def minusYears(years: Long): YearMonth = minus(years, YEARS)
+
+  def minusMonths(months: Long): YearMonth = minus(months, MONTHS)
+
+  // Not implemented
+  // def query[R](query: TemporalQuery[R]): R
+
+  def adjustInto(temporal: Temporal): Temporal =
+    temporal.`with`(PROLEPTIC_MONTH, prolepticMonth)
+
+  def until(end: Temporal, unit: TemporalUnit): Long = {
+    val other = YearMonth.from(end)
+    val monthsDiff = other.prolepticMonth - prolepticMonth
+    unit match {
+      case MONTHS    => monthsDiff
+      case YEARS     => monthsDiff / 12
+      case DECADES   => until(end, YEARS) / 10
+      case CENTURIES => until(end, YEARS) / 100
+      case MILLENNIA => until(end, YEARS) / 1000
+      case ERAS      => other.getLong(ERA) - getLong(ERA)
+
+      case _: ChronoUnit =>
+        throw new UnsupportedTemporalTypeException("Unsupported unit: " + unit)
+
+      case _ => unit.between(this, end)
+    }
+  }
+
+  def atDay(dayOfMonth: Int): LocalDate =
+    LocalDate.of(year, month, dayOfMonth)
+
+  def atEndOfMonth: LocalDate =
+    atDay(getMonth().length(isLeapYear()))
+
+  def compareTo(other: YearMonth): Int = {
+    if (year == other.getYear)
+      month - other.getMonthValue
+    else
+      year - other.getYear
+  }
+
+  def isAfter(other: YearMonth): Boolean = compareTo(other) > 0
+
+  def isBefore(other: YearMonth): Boolean = compareTo(other) < 0
+
+  override def equals(other: Any): Boolean = other match {
+    case that: YearMonth =>
+      year == that.getYear && month == that.getMonthValue
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(year, month)
+    state.foldLeft(0)((a, b) => (a * 31) + b)
```

Comment:
After experimenting a bit with the hashcode on the JVM, it seems that it is implemented as `year + (month << 27)`.